addons/udes_stock/tests/test_picking.py

```diff
@@ -167,3 +167,213 @@ def test12_package_has_no_parent_package(self):
         _, pick = self.generate_picks_and_pallets_for_check_entire_pack()
         pick._check_entire_pack()
         self.assertFalse(pick.move_line_ids.u_result_parent_package_id)
+
+class TestSuggestedLocation(common.BaseUDES):
+
+    @classmethod
+    def setUpClass(cls):
+        """Setup test data to test suggested locations."""
+        super(TestSuggestedLocation, cls).setUpClass()
+
+        Location = cls.env['stock.location']
+        Picking = cls.env['stock.picking']
+
+        cls.test_location_03 = Location.create({
+            "name": "Test location 03",
+            "barcode": "LTEST03",
+            "location_id": cls.stock_location.id,
+        })
+        cls.test_location_04 = Location.create({
+            "name": "Test location 04",
+            "barcode": "LTEST04",
+            "location_id": cls.stock_location.id,
+        })
+        cls.test_locations += cls.test_location_03 + cls.test_location_04
+
+        cls.test_stock_quant_01 = cls.create_quant(
+            cls.tangerine.id,
+            cls.test_location_01.id,
+            10,
+            "TESTLOT001"
+        )
+        cls.test_stock_quant_02 = cls.create_quant(
+            cls.tangerine.id,
+            cls.test_location_02.id,
+            10,
+            "TESTLOT002"
+        )
+        # Create a non-lot tracked quant
+        cls.test_stock_quant_03 = cls.create_quant(
+            cls.apple.id,
+            cls.test_location_03.id,
+            10,
+        )
+
+        # Create a new lot tracked product
+        cls.uglyfruit = cls.create_product('Ugly Fruit', tracking='lot')
+
+    def create_and_assign_putaway_picking(
+        self,
+        products_info,
+        drop_policy="exactly_match_move_line"
+    ):
+        """Create and assign a putaway picking with associated quants created."""
+        for info in products_info:
+            test_quant = self.create_quant(
+                info["product"].id,
+                self.received_location.id,
+                info["qty"],
+            )
+            if info.get("lot"):
+                test_quant.lot_id = info["lot"]
+            # Remove lot from dictionary (if present) so that it may be used in create_picking
+            info.pop("lot", None)
```

Comment:
We can move this line to before line 227:
`lot = info.php("lot", None)`
And then check `if lot:` instead of get.

Reply:
Good spot